added fixed navbar

nathanchi opened the pull request.

sushain97 commented on source-browser.html:
```diff
@@ -10,8 +10,35 @@
     <link rel="stylesheet" href="https://unpkg.com/basscss@7.1.1/css/basscss.min.css" integrity="sha384-/biuPsPEkt10QoeOExIADuWNrpFFXALKPgBhEDkRdKDJdQOPpbNU3uUeFbL/KPJg" crossorigin="anonymous">
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/hint.css/2.5.0/hint.base.min.css" integrity="sha256-ODFgZG+Y3V10lzZu+7J8epM4SdW0w3p8qbVmmShNrtk=" crossorigin="anonymous">
     <style type="text/css">
+
```
Remove extra line.

sushain97 commented on source-browser.html:
```diff
@@ -42,6 +69,16 @@
   </head>
   <body class="m2">
+    <ul class = "navbar" id="navbar">
+      <li><a href="#languages" id="lilang">languages</a></li>
```
These should not be hardcoded. They should be generated by the JS.

sushain97 commented on source-browser.html:
```diff
@@ -72,16 +109,16 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
           repoUrl: `https://github.com/${ORGANIZATION}/apertium-${topic}`,
         }));
-
+        
```
Please remove _all_ these extra trailing spaces.

sushain97 replied:
These are not fixed...

nathanchi replied:
I have removed these from throughout the code.

sushain97 commented on source-browser.html:
```diff
@@ -132,7 +169,7 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
             $('<div class="topic flex-auto">').append(
               $('<div>').append(
-                $('<h2 class="inline-block m0">').text(name),
+                $('<h2 class="inline-block m0" id='+name+'>').text(name),
```
Don't use concatenation. Use a jQuery function to set the `id`.

sushain97 commented on source-browser.html:
```diff
@@ -166,45 +203,129 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+        $(window).scroll(function() {
+            var top = $(window).scrollTop();
+            var bottom = $(window).scrollTop() + $(window).height();
+            var langtop = $("#languages").position(); 
```
These should be adjusted via a loop that goes through the topics, not hardcoded to only work with these.

sushain97 commented on source-browser.html:
```diff
@@ -42,6 +65,7 @@
     </style>
   </head>
   <body class="m2">
+    <ul class = "navbar" id="navbar"></ul>
```
Remove spacing around first equal sign.

sushain97 commented on source-browser.html:
```diff
@@ -166,28 +191,57 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
-
+        
+        
+        var array = ['languages', 'trunk', 'staging', 'nursery', 'incubator', 'tools'];
```
This is already defined above as TOPICS.

sushain97 commented on source-browser.html:
```diff
@@ -166,28 +191,57 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+        
+        var array = ['languages', 'trunk', 'staging', 'nursery', 'incubator', 'tools'];
+        array.forEach(function(header) {
+          $("#navbar").append("<li><a href=#"+header+" id=li"+header+">"+header+"</a></li>");
```
There should be no concatenation here.

sushain97 replied:
Try using the href and text methods. jQuery elements can also take a list of attributes. Nested elements should be constructed via append for readability.

sushain97 commented on source-browser.html:
```diff
@@ -166,28 +191,57 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+        var array = ['languages', 'trunk', 'staging', 'nursery', 'incubator', 'tools'];
+        array.forEach(function(header) {
+          $("#navbar").append("<li><a href=#"+header+" id=li"+header+">"+header+"</a></li>");
+          $("#inline-block m0").attr("id", name);
```
What element is this adding to? Why is there an element with this misleading id that sounds like a class?

sushain97 commented on source-browser.html:
```diff
@@ -166,28 +191,57 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+        
+        
+        var array = ['languages', 'trunk', 'staging', 'nursery', 'incubator', 'tools'];
+        array.forEach(function(header) {
```
This should all be happening inside the DOM ready callback.

sushain97 commented on source-browser.html:
```diff
@@ -166,28 +191,57 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+          $("#navbar").append("<li><a href=#"+header+" id=li"+header+">"+header+"</a></li>");
+          $("#inline-block m0").attr("id", name);
+        });
+        $(window).scroll(function() {
```
Please use only arrow functions for anonymous functions.

sushain97 commented on source-browser.html:
```diff
@@ -166,28 +191,57 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+          $("#inline-block m0").attr("id", name);
+        });
+        $(window).scroll(function() {
+            var top = $(window).scrollTop();
```
Use only const or let, never var.

sushain97 commented on source-browser.html:
```diff
@@ -166,28 +191,57 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+            var bottom = $(window).scrollTop() + $(window).height();
+            var titles = ['languages', 'trunk', 'staging', 'nursery', 'incubator', 'tools'];
+            var i;
+            for(i = 0; i<titles.length; i++){
```
Need spacing between binary operators and i should be declared inline here.

sushain97 commented on source-browser.html:
```diff
@@ -166,28 +191,57 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+        $(window).scroll(function() {
+            var top = $(window).scrollTop();
+            var bottom = $(window).scrollTop() + $(window).height();
+            var titles = ['languages', 'trunk', 'staging', 'nursery', 'incubator', 'tools'];
```
Why is this declared again?

sushain97 commented on source-browser.html:
```diff
@@ -166,28 +191,57 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+            for(i = 0; i<titles.length; i++){
+                var header = titles[i];
+                var vartop = $('#'+header).position();
+                  if(top < vartop.top && vartop.top < bottom){
```
Please emulate spacing in the rest of this code.

sushain97 commented on source-browser.html:
```diff
@@ -166,28 +191,57 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+                      if(vartop.top>bottom){
+                          $tools.text(header+'↓')
+                      }
+                      else
```
Always use braces.

sushain97 commented on source-browser.html:
```diff
@@ -67,21 +91,21 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
         EXPIRY_KEY = `${ORGANIZATION}-expiry`,
         EXPIRY_MILLIS = 60000,
-        TOPICS = ['languages', 'trunk', 'staging', 'nursery', 'incubator', 'tools'].map(topic => ({
+        LIST = ['languages', 'trunk', 'staging', 'nursery', 'incubator', 'tools'],
```
Please don't use entirely unhelpful variables names like `list`. In fact, this change isn't necessary at all. Just use the previous `TOPICS` variable for iteration and access the `name` attribute. e.g. `TOPICS[0].name`.

sushain97 commented on source-browser.html:
```diff
@@ -67,21 +91,21 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
         const headers = new Headers({ Accept: 'application/vnd.github.mercy-preview+json' });
         const cacheStale = !(CACHE_KEY in localStorage) || localStorage[EXPIRY_KEY] < Date.now();
-
+          
```
There are still unnecessary whitespace additions throughout this entire file that should be deleted. Something like 10 of them.

sushain97 commented on source-browser.html:
```diff
@@ -132,7 +156,7 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
             $('<div class="topic flex-auto">').append(
               $('<div>').append(
-                $('<h2 class="inline-block m0">').text(name),
+                $('<h2 class="inline-block m0" id="topic" style="padding-top:75px; margin-top:-75px;">').text(name),
```
Don't use inline styling. Instead, style the associated class. Furthermore, this creates multiple elements with the same id, i.e. `topic`. This is illegal HTML. Instead, something like `topic-languages` etc. might work for your purposes.

sushain97 commented on source-browser.html:
```diff
@@ -141,7 +165,8 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
             ),
           );
-
+          $("[id=topic]").attr("id", name);
```
See above comment.

sushain97 commented on source-browser.html:
```diff
@@ -166,28 +191,60 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+                  else{
+                      $('#li'+heading).show();
+                      let $heading = $('#li'+heading);
+                      if(vartop.top>bottom){
```
Spacing between binary operations here and throughout should be consistent with existing code.

sushain97 commented on source-browser.html:
```diff
@@ -166,28 +191,60 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+            
       $(document).ready(() => {
+        $("#navbar").append('<li id="li"></li>');
+        for(let i = 0; i < LIST.length; i++){ 
```
For the sake of stylistic consistency, use `forEach` here. It might even be nicer with a `map` to create the elements and then appending the entire list to the container.

sushain97 commented on source-browser.html:
```diff
@@ -166,28 +191,60 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+        $("#navbar").append('<li id="li"></li>');
+        for(let i = 0; i < LIST.length; i++){ 
+            $("#li").append('<a id="placeholder"></a>');
+            $("[id=placeholder]").attr({
```
This is an anti-pattern. Use the way that multiple attributes are added to an element throughout the existing code, i.e. an object containing the attributes is passed as the second argument to `$(...)` when creating the element.

sushain97 commented on source-browser.html:
```diff
@@ -201,10 +258,10 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
     </script>
   </body>
-</html>
+</html>
```
Please revert your changes that deleted the final blank line in this file.

sushain97 commented on source-browser.html:
```diff
@@ -10,8 +10,35 @@
+          padding: 14px 16px;
+          text-decoration: none;
+      }
+      h2[class="inline-block m0"] {
```
Please look into how to use CSS selectors, especially selecting using classes.

nathanchi replied:
Okay, I'll take a look at that.

nathanchi replied:
I have fixed the CSS.

nathanchi replied:
I'll fix the spacing.

sushain97 commented on source-browser.html:
```diff
@@ -181,13 +209,51 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+      function createNavbar(item) {
+          $('#li').append(
+              $('<a>', {
+                  id: 'li' + item,
```
Use string interpolation instead of concatenation i.e. 
    
    `li${item}`

Also, `id`s should be id-agnostic if possible. So, something like `${item}-nav-link` perhaps?

sushain97 commented on source-browser.html:
```diff
@@ -181,13 +209,51 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
           .focus();
       }
 
+      function createNavbar(item) {
```
```suggestion
      function createNavbar(topic) {
```

sushain97 commented on source-browser.html:
```diff
@@ -181,13 +209,51 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+          for(let i = 0; i < TOPICS.length; i++){
+              let heading = TOPICS[i].name;
+              list.push(heading);
+              }
```
Indentation is off here.

sushain97 commented on source-browser.html:
```diff
@@ -181,13 +209,51 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
       $(document).ready(() => {
+        let list = [];
+        $("#navbar").append('<li id="li"></li>');
+          for(let i = 0; i < TOPICS.length; i++){
```
Use a `forEach` instead for consistency.

sushain97 commented on source-browser.html:
```diff
@@ -181,13 +209,51 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
           currentTarget.blur();
           refreshTopics();
         });
-
```
Revert spacing change here.

sushain97 commented on source-browser.html:
```diff
@@ -181,13 +209,51 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+              let heading = TOPICS[i].name;
+              let vartop = $('#' + heading).position();
+                if(top < vartop.top && vartop.top < bottom){
+                    $('#li' + heading).hide();
```
Don't use string interpolation.

sushain97 commented on source-browser.html:
```diff
@@ -181,13 +209,51 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+          for(let i = 0; i < TOPICS.length; i++){
+              let heading = TOPICS[i].name;
+              let vartop = $('#' + heading).position();
+                if(top < vartop.top && vartop.top < bottom){
```
```suggestion
                if (top < vartop.top && vartop.top < bottom) {
```

sushain97 commented on source-browser.html:
```diff
@@ -181,13 +209,51 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+      $(window).scroll(() => {
+          let top = $(window).scrollTop();
+          let bottom = $(window).scrollTop() + $(window).height();
+          for(let i = 0; i < TOPICS.length; i++){
```
This should be possible with `forEach`. Don't create a new function, just use an anonymous arrow function.

sushain97 commented on source-browser.html:
```diff
@@ -181,13 +209,51 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+                if(top < vartop.top && vartop.top < bottom){
+                    $('#li' + heading).hide();
+            }
+                else{
```
Spacing and indentation

sushain97 commented on source-browser.html:
```diff
@@ -181,13 +209,51 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+                else{
+                    $('#li' + heading).show();
+                    let $heading = $('#li' + heading);
+                    if(vartop.top > bottom){
```
```suggestion
                    if (vartop.top > bottom) {
```

sushain97 commented on source-browser.html:
```diff
@@ -10,8 +10,35 @@
+          padding: 14px 16px;
+          text-decoration: none;
+      }
+      .inline-block.m0 {
```
Should not be styling all of them with this. Instead, should use a custom class in the JS and style that.

sushain97 commented on source-browser.html:
```diff
@@ -10,8 +10,35 @@
+      ul:nth-child(1) {
+          list-style-type: none;
+          position: absolute;
+          margin: 0;
```
Many of these rules should be replaced with the basscss v7 rules. e.g. `.m0` is `margin: 0`.

sushain97 commented on source-browser.html:
```diff
@@ -42,6 +69,7 @@
     </style>
   </head>
   <body class="m2">
+    <ul class="navbar" id="navbar"></ul>
```
Just the `id` should suffice unless you have multiple `navbar`s.

sushain97 commented on source-browser.html:
```diff
@@ -181,7 +209,46 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
           .focus();
       }
 
+      function createNavbar(topic) {
```
Inline this function where it is used.

sushain97 commented on source-browser.html:
```diff
@@ -181,7 +209,46 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+      function createNavbar(topic) {
+        $('.li').append(
+          $('<a>', {
+            id: `li${topic}`,
```
As I mentioned in my previous review, use a tag-agnostic id.

sushain97 commented on source-browser.html:
```diff
@@ -181,7 +209,46 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+        );
+      }
+
+      let list = [];
```
As I mentioned earlier, variable names that are just the type are undesirable in most languages unless it's a utility function with no context.

sushain97 commented on source-browser.html:
```diff
@@ -181,7 +209,46 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+      }
+
+      let list = [];
+      $(document).ready(() => {
```
Please don't use multiple DOM ready calls. This can just be added to the one below.

nathanchi replied:
I have fixed that.

sushain97 commented on source-browser.html:
```diff
@@ -181,7 +209,46 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+
+      let list = [];
+      $(document).ready(() => {
+	    TOPICS.forEach(item => {
```
```suggestion
	    TOPICS.forEach(({name}) => {
```

sushain97 commented on source-browser.html:
```diff
@@ -181,7 +209,46 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+      $(document).ready(() => {
+	    TOPICS.forEach(item => {
+	      let heading = item.name;
+	      list.push(heading);
```
This `list` seems completely unnecessary. `TOPICS` can be used directly below instead.

nathanchi replied:
What method for TOPICS should I use?  TOPICS. push(heading) doesn't work.

nathanchi replied:
What method for TOPICS should I use?  TOPICS. push(heading) doesn't work.

sushain97 replied:
`TOPICS.forEach` can be used directly below.

nathanchi replied:
Sorry, I meant 

TOPICS.name.forEach(topic => {

doesn't work.

sushain97 replied:
It can work if you use topic.name later.

nathanchi replied:
That makes sense.   When I do TOPICS.forEach, though, it just gives [object Object] as the footer item names.

nathanchi replied:
![screen shot 2018-11-17 at 12 40 07 pm](https://user-images.githubusercontent.com/44486178/48665579-f6eb1c80-ea65-11e8-923b-95ea6fdb5f67.png)

sushain97 replied:
Try using the debugger.

nathanchi replied:
Okay, I managed to get some of it working—the titles are there now, but the arrows are gone.

sushain97 replied:
Debugger :)

nathanchi replied:
It works now!

sushain97 commented on source-browser.html:
```diff
@@ -181,7 +209,46 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+	    });
+      });
+
+      $(window).scroll(() => {
```
Should this also happen on window resize?

nathanchi replied:
I've added that, as well.

sushain97 commented on source-browser.html:
```diff
@@ -181,7 +209,46 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+        let top = $(window).scrollTop();
+        let bottom = $(window).scrollTop() + $(window).height();
+        list.forEach(heading => {
+          let vartop = $(`#${heading}`).position();
```
What does the `var` in `vartop` refer to?

nathanchi replied:
It refers to variable, for each of the heading variables.  Would to suggest I rename it?

sushain97 replied:
Does it make sense to name a variable `var` to indicate that it's a variable? Something like `heading_position` maybe.

nathanchi replied:
Okay, I'll rename that.

sushain97 commented on source-browser.html:
```diff
@@ -181,7 +209,46 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+      });
+
+      $(window).scroll(() => {
+        let top = $(window).scrollTop();
```
Variables that do not change should be `const`.

sushain97 commented on source-browser.html:
```diff
@@ -181,7 +209,46 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+            $(`#li${heading}`).hide();
+          } else {
+            $(`#li${heading}`).show();
+            let $heading = $(`#li${heading}`);
```
```suggestion
            $(`#li${heading}`).text(`${heading} ${vartop.top > bottom ? '↓' : '↑'}`);
```

sushain97 commented on source-browser.html:
```diff
@@ -181,7 +209,46 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+      });
+
       $(document).ready(() => {
+        $('#navbar').append('<li class="li"></li>');
```
Why is there an extra list element?

nathanchi replied:
There's only one list element.  Would you like me to make this list element hard-coded in the html instead?

sushain97 replied:
Yes

nathanchi replied:
Great, I'll do that. :)

nathanchi commented:
I'm using my dad's computer.  That's why the name is different.

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,32 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+            const top = $(window).scrollTop();
+            const bottom = $(window).scrollTop() + $(window).height();
+            TOPICS.forEach(heading => {
+              heading = heading.name;
```
Use destructuring in the function argument.

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,32 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+            const bottom = $(window).scrollTop() + $(window).height();
+            TOPICS.forEach(heading => {
+              heading = heading.name;
+              let heading_position = $(`#${heading}`).position();
```
Const

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,32 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+            TOPICS.forEach(heading => {
+              heading = heading.name;
+              let heading_position = $(`#${heading}`).position();
+              if (top < heading_position.top && heading_position.top < bottom) {
```
If you're only ever using top, just save that instead.

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,32 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
       $(document).ready(() => {
+        $(window).on('scroll resize', function(){
+            const top = $(window).scrollTop();
+            const bottom = $(window).scrollTop() + $(window).height();
```
Use top here instead of another call.

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,32 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+                $(`#${heading}-nav-link`).hide();
+              } else {
+                $(`#${heading}-nav-link`).show();
+                $(`#${heading}-nav-link`).text(`${heading} ${heading_position.top > bottom ? '↓' : '↑'}`);
```
These calls can be chained.

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,32 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+          });
+
+        TOPICS.forEach(topic => {
+            topic = topic.name;
```
Destructuring here as well.

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,32 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+            topic = topic.name;
+            $('.li').append(
+              $('<a>', {
+                class: `inline-block white center btn:hover`,
```
I would just add this to the string part since there's no interpolation.

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,32 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+        $(window).on('scroll resize', function(){
+            const top = $(window).scrollTop();
+            const bottom = $(window).scrollTop() + $(window).height();
+            TOPICS.forEach(heading => {
```
I think append can take a list, right? In that case, use a map to create the list and append that.

nathanchi replied:
Which append are you referring to?  
 `$('.li').append(` ?

nathanchi replied:
From what I understand, that area of the code doesn't need appending.

nathanchi commented on source-browser.html:
```diff
@@ -42,6 +50,9 @@
   </head>
   <body class="m2">
+    <ul class="fixed m0 p0 overflow-hidden bg-black bottom-0 col-12 z2" id="navbar">
+      <li class="li left"></li>
```
This is the hardcoded li class that I added.  The <a> goes within the <li>, which goes within the <ul>.

nathanchi replied:
It's used for the .append() method later on in the code, to select only those values.

sushain97 replied:
Why not just `#navbar li`?.

sushain97 replied:
Unresolved?

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,29 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+        $(window).on('scroll resize', function(){
+            const top = $(window).scrollTop();
+            const bottom = top + $(window).height();
+            TOPICS.forEach(topic => {
```
```suggestion
            TOPICS.forEach(({name}) => {
```

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,29 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+            const bottom = top + $(window).height();
+            TOPICS.forEach(topic => {
+              let {name} = topic;
+              const name_position = $(`#${name}`).position().top;
```
```suggestion
              const navLinkPosition = $(`#${name}`).position().top;
```

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,29 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
 
       $(document).ready(() => {
+        $(window).on('scroll resize', function(){
+            const top = $(window).scrollTop();
```
The rest of the code is indented two spaces but now it's 4?

nathanchi replied:
I'll run it through Prettier again, then.

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,29 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
       }
 
       $(document).ready(() => {
+        $(window).on('scroll resize', function(){
```
```suggestion
        $(window).on('scroll resize', () => {
```

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,29 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+              if (top < name_position && name_position < bottom) {
+                $(`#${name}-nav-link`).hide();
+              } else {
+                $(`#${name}-nav-link`).text(`${name} ${name_position > bottom ? '↓' : '↑'}`).show();
```
Can replace the entire block with something like:

```suggestion
                $(`#${name}-nav-link`)
                  .text(`${name} ${name_position > bottom ? '↓' : '↑'}`)
                  .toggle(top >= name_position || name_position >= bottom);
```

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,29 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+              }
+            });
+          });
+        TOPICS.forEach(topic => {
```
This entire block can be replaced with `$('.li').append(TOPICS.map(({name}) => ...`

sushain97 replied:
Also, why are you appending all the links to the same list element? There should be one list element (`<li>`) per entry inside the `<ul>`.

nathanchi replied:
I tried replacing TOPICS.forEach(topic => { as you recommended, but it didn't work.

I also tried to make one list element (each containing one /<a> element), but it didn't quite work.  I tried doing this:

`          TOPICS.map(({name}) => {
            $('#navbar').append(
			  $('<li></li>', {class: `${name} left`})
              $('#navbar li').append(
                $('<a class = "inline-block white center btn:hover">', {
                  id: `${name}-nav-link`,
                  href: `#${name}`
                }).text(name)
              );
		    );
         });`, which failed.  Do you have any suggestions?

sushain97 commented on source-browser.html:
```diff
@@ -182,20 +192,43 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+              $('<a class = "inline-block white center btn:hover">', {
+                id: `${name}-nav-link`,
+                href: `#${name}`,
+                class: `inline-block white center btn:hover`
```
You already have it above in the string.

nathanchi replied:
On 200?  It doesn't define the id, href, or class.

nathanchi replied:
Oh, I'm sorry, I misunderstood what you meant.

nathanchi commented on source-browser.html:
```diff
@@ -182,20 +192,43 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+        });
+
+        TOPICS.map(({name}) => {
+          $('#navbar').append(
```
So something like this? 
`        $('#navbar').append(
            TOPICS.map(({ name }) => {`

nathanchi replied:
The js syntax seems to not work when I do that, however.  Do you have any suggestions how to do this?

nathanchi replied:
Do you know what the error might be?

sushain97 replied:
You shouldn't have multiple semicolons in one statement.

nathanchi replied:
I edited the jquery as you suggested, and now the code looks like this:

https://pastebin.com/T1RPg36n

It still doesn't work, though...I'm pretty sure the syntax is correct, though.  Every parentheses/bracket is balanced and closed.

sushain97 replied:
Please just paste code here using the code formatting (4 spaces).

I don't understand why you have two lines that are almost identical? Also, your arrow function needs a return.

nathanchi replied:
Alright, I've fixed it.

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +192,27 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+			}).text(name)
+			),
+		  ),
+		);
```
Spacing is very suspicious here. Try running this through Prettier.

nathanchi commented:
Nevermind, I got it to work!  :)

nathanchi commented:
I fixed everything, but there's something wrong about the anchor links: any header that's on the top of the page doesn't work, for some reason.  Do you have any ideas about this?  I fixed everything else, though.

nathanchi commented:
Currently, I'm using

`      .heading {`
`        padding-top: 75px;`
`        margin-top: -75px;`
`    }`

which works for all of the other links besides the ones on the top.  I also tried using some of the solutions from here: https://css-tricks.com/hash-tag-links-padding/ but it didn't quite work...

nathanchi commented:
I think that there might be something wrong with the header (with the apertium & github logos), which is making the text headers below messed up.

sushain97 commented on source-browser.html:
```diff
@@ -10,8 +10,19 @@
+      #navbar li a {
+        padding: 14px 16px;
+      }
+	  .heading {
```
Indentation is off in this entire block.

sushain97 commented on source-browser.html:
```diff
@@ -10,8 +10,19 @@
       #content {
+        margin: 1rem;
         margin-top: 75px;
+        margin-bottom: 100px;
```
Instead of using 3 margin rules, just use one with 4 values.

sushain97 commented on source-browser.html:
```diff
@@ -41,8 +52,10 @@
   </head>
-  <body class="m2">
-    <div class="fixed top-0 col-12 py1 mr3 bg-white z1">
+  <body class="">
```
Please remove the empty class attribute.

sushain97 commented on source-browser.html:
```diff
@@ -41,8 +52,10 @@
-  <body class="m2">
-    <div class="fixed top-0 col-12 py1 mr3 bg-white z1">
+  <body class="">
+    <ul id="navbar" class="fixed m0 p0 overflow-hidden bg-black bottom-0 col-12 z2">
```
```suggestion
    <ul id="navbar" class="fixed m0 p0 overflow-hidden bg-black bottom-0 col-12 z2"></ul>
```

sushain97 replied:
(i.e. on the same line)

sushain97 commented on source-browser.html:
```diff
@@ -132,7 +145,8 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
             $('<div class="topic flex-auto">').append(
               $('<div>').append(
-                $('<h2 class="inline-block m0">').text(name),
+                $('<h2 class="inline-block m0 relative heading">', {id: `${name}-space`}),
```
Why not just `id: name` like before? The `space` suffix seems odd.

nathanchi replied:
I need to use a different id for the empty space above the h2 header, that's why I'm using name-space.

sushain97 commented on source-browser.html:
```diff
@@ -10,8 +10,17 @@
+        display: block;
+        top: -90px;
+        visibility: hidden;
+	  }
```
Still off here?

sushain97 commented on source-browser.html:
```diff
@@ -59,7 +69,7 @@
       <div id="topics" class="flex flex-wrap col-12 mt1"></div>
     </div>
-    <a class="fixed right-0 bottom-0 mr2 mb2 text-decoration-none h2" id="refresh" href="#">🔄</a>
+    <a class="fixed right-0 bottom-0 mr2 mb2 text-decoration-none h2 z3" id="refresh" href="#">🔄</a>
```
Can we just move it up so it doesn't sit on top of the navbar?

sushain97 replied:
Or, center it vertically in the navbar.

sushain97 commented on source-browser.html:
```diff
@@ -41,8 +50,9 @@
-  <body class="m2">
-    <div class="fixed top-0 col-12 py1 mr3 bg-white z1">
+  <body>
+    <ul id="navbar" class="fixed m0 p0 overflow-hidden bg-black bottom-0 col-12 z2"></ul>
```
This is pretty dark. Maybe something like `#2e2e2e`.

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,27 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+        $('#navbar').append(
+          TOPICS.map(({ name }) =>
+            $('<li class="left">', {id: `${name}-nav-list`}).append(
+              $('<a class="inline-block white center btn:hover">', {
```
Can we remove the text underline on these? It's quite odd. Maybe use some of the styles from this page's http://basscss.com/v7/docs/guides/ui/#! Navigation section?

sushain97 commented on source-browser.html:
```diff
@@ -132,7 +142,8 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
             $('<div class="topic flex-auto">').append(
               $('<div>').append(
-                $('<h2 class="inline-block m0">').text(name),
+                $('<h2 class="inline-block m0 relative topic-heading">', {id: `${name}-space`}),
```
Wait. I didn't realize you had two of these elements... This is super hacky :( Have you tried either of the solutions at https://stackoverflow.com/questions/4086107/fixed-page-header-overlaps-in-page-anchors ?

nathanchi replied:
I have, actually.   They haven't worked especially well: the main problem is that the headers at the top don't work.

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,27 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
       }
 
       $(document).ready(() => {
+        $(window).on('scroll resize load', () => {
```
Let's pull the anonymous function here into a function called `updateNavbarVisibility`.

sushain97 replied:
No, I didn't mean the whole thing, I meant just the arrow function.

nathanchi replied:
Okay, I'll fix that.

nathanchi replied:
So just `          TOPICS.forEach(({ name }) => {
            const navLinkPosition = $(`#${name}`).position().top;
            $(`#${name}-nav-link`).text(`${name} ${navLinkPosition > bottom ? '↓' : '↑'}`);
            $(`#${name}-nav-list`).toggle(top >= navLinkPosition || navLinkPosition >= bottom);		
          });`?

sushain97 replied:
As I mentioned earlier, please use code blocks via 4 space indents.

    $(window).on('scroll resize load', updateNavbarVisibility)

is what I'm looking for.

sushain97 commented on source-browser.html:
```diff
@@ -182,6 +193,27 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+          });
+        });
+
+        $('#navbar').append(
```
Let's move this into a function called `drawNavbar()`.

nathanchi commented:
I accidentally resolved the conversation...I have tried the solutions at https://stackoverflow.com/questions/4086107/fixed-page-header-overlaps-in-page-anchors, but they didn't entirely work.  The hacky solution was the only solution that has worked entirely, so far.

nathanchi commented:
These are the CSS solutions.

sushain97 commented on source-browser.html:
```diff
@@ -181,7 +205,34 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+
       $(document).ready(() => {
+        updateNavbarVisibility();
+        drawNavbar();
```
Newline after this.

nathanchi replied:
Okay, I'll add that.

sushain97 commented on source-browser.html:
```diff
@@ -181,7 +205,34 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+          $(`#${name}-nav-link`).text(`${name} ${navLinkPosition > bottom ? '↓' : '↑'}`);
+          $(`#${name}-nav-list`).toggle(top >= navLinkPosition || navLinkPosition >= bottom);		
+        });
+      };
```
There should not be a semicolon here.

sushain97 commented on source-browser.html:
```diff
@@ -181,7 +205,34 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+        });
+      };
+      
+
```
There's an extra newline here.

sushain97 commented on source-browser.html:
```diff
@@ -181,7 +205,34 @@ <h1 id="loading" class="m1 center display-none">⌛</h1>
+
       $(document).ready(() => {
+        $(window).on('scroll resize load', updateNavbarVisibility);
+        drawNavbar();
```
Switch these two lines just so that it makes more sense semantically.

sushain97 commented:
for future readers: blocking on merge till I can look into whether there's a cleaner solution to offsetting the anchor jump location
